spin1_api/spin1_api.c

```diff
@@ -773,18 +774,19 @@ static void report_warns(void)
 }
 
 
+#ifdef __GNUC__
+    register uint _lr asm("lr");
```

Comment:
At the very least, this should be `static register` as the _declaration_ should not be exported by the file. But it's weird to have this at the top level scope of the file; was there a reason for this?

Reply:
This is a rather special declaration in that it is referencing an actual register, and you don't want to change the value currently there.  Apparently you can't have a "static register", maybe because of this.  The register exists globally, so it doesn't really matter where you declare it, but putting it here stops the arm compiler complaining that you "might not have assigned it a value", when of course you don't want to assign it a *new* value, but read the existing value!

Reply:
OK. It's weird, but the compiler complaining about an unassigned register-bound variable is also weird.